File: src/main/java/me/skynda/vehicle/entity/VehicleModel.java
package me.skynda.vehicle.entity;

import javax.persistence.*;

import lombok.Data;

import java.io.Serializable;
import java.util.Date;

@Entity
@Data
@Table(name = "vehicle_model")
public class VehicleModel implements Serializable{

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "id", columnDefinition = "serial")
	private Long id;

	@Column(name = "model_code")
	private String modelCode;

	@Column(name = "description")
	private String description;

	@Column(name = "title")
	private String title;

	@Column(name = "engine")
	private String engine;

	@Column(name = "horse_power")
	private Integer horsePower;

	@Column(name = "doors")
	private Integer doors;

	@Column(name = "seats")
	private Integer seats;

	@Column(name = "year")
	private Integer year;

	@Column(name = "archived")
	private Date archived;

	@ManyToOne(fetch = FetchType.LAZY, optional = false)
	@JoinColumn(name = "vehicle_manufacturer_id", nullable = false)
	private Classification vehicleManufacturer;

	@ManyToOne(fetch = FetchType.LAZY, optional = false)
	@JoinColumn(name = "transmission_id", nullable = false)
	private Classification transmission;

	@ManyToOne(fetch = FetchType.LAZY, optional = false)
	@JoinColumn(name = "drivetrain_id", nullable = false)
	private Classification drivetrain;

	@ManyToOne(fetch = FetchType.LAZY, optional = false)
	@JoinColumn(name = "vehicle_body_id", nullable = false)
	private Classification vehicleBody;

	@ManyToOne(fetch = FetchType.LAZY, optional = false)
	@JoinColumn(name = "fuel_type_id", nullable = false)
	private Classification fuelType;
}

File: skynda_dotnet_backend/Triven.Data.EntityFramework/Models/VehicleMOdel.cs
using System.ComponentModel.DataAnnotations.Schema;
using Triven.Data.EntityFramework.Models.Base;
using Triven.Domain.Models;

namespace Triven.Data.EntityFramework.Models
{
    [Table("VehicleModel")]
    public class VehicleModel : AuditableModel, IVehicleModel
    {
        public string Description { get; set; }

        public string Title { get; set; }

        public int HorsePower { get; set; }

        public int Doors { get; set; }

        public int Seats { get; set; }

        public string ModelCode { get; set; }

        public string Engine { get; set; }

        public int Year { get; set; }   

        /// <summary>
        /// Manufacturer
        /// </summary>
        public int VehicleManufacturerId { get; set; }
        public virtual Classification VehicleManufacturer { get; set; }

        /// <summary>
        /// Transmission
        /// </summary>
        public int TransmissionId { get; set; }
        public virtual Classification Transmission { get; set; }

        /// <summary>
        /// Drivetrain
        /// </summary>
        public int DrivetrainId { get; set; }
        public virtual Classification Drivetrain { get; set; }

        /// <summary>
        /// VehicleBody
        /// </summary>
        public int VehicleBodyId { get; set; }
        public virtual Classification VehicleBody { get; set; }

        /// <summary>
        /// Fuel
        /// </summary>
        public int FuelTypeId { get; set; }
        public virtual Classification FuelType { get; set; }
    }
}